flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java

```diff
@@ -0,0 +1,763 @@
+		assertVertexInState(ExecutionState.DEPLOYING, ev11);
+		assertVertexInState(ExecutionState.FINISHED, ev12);
+		assertVertexInState(ExecutionState.CREATED, ev21);
+		assertVertexInState(ExecutionState.FINISHED, ev22);
+		assertVertexInState(ExecutionState.CREATED, ev31);
+		assertVertexInState(ExecutionState.CREATED, ev32);
+		assertVertexInState(ExecutionState.CREATED, ev41);
+		assertVertexInState(ExecutionState.CREATED, ev42);
+
+		// verify attempt number
+		assertEquals(1, ev11.getCurrentExecutionAttempt().getAttemptNumber());
+		assertEquals(0, ev12.getCurrentExecutionAttempt().getAttemptNumber());
+		assertEquals(1, ev21.getCurrentExecutionAttempt().getAttemptNumber());
+		assertEquals(0, ev22.getCurrentExecutionAttempt().getAttemptNumber());
+		assertEquals(1, ev31.getCurrentExecutionAttempt().getAttemptNumber());
+		assertEquals(1, ev32.getCurrentExecutionAttempt().getAttemptNumber());
+		assertEquals(1, ev41.getCurrentExecutionAttempt().getAttemptNumber());
+		assertEquals(1, ev42.getCurrentExecutionAttempt().getAttemptNumber());
+	}
+
+	/**
+	 * Tests that checkpointing is working correctly in region failover.
+	 */
+	@Test
+	public void testCheckpointingInRegionFailover() throws Exception {
+
+		final JobGraph jobGraph = createStreamingJobGraph();
+		final long checkpointId = 42L;
+		final ExecutionGraph eg = createExecutionGraph(jobGraph, checkpointId);
+
+		final Iterator<ExecutionVertex> vertexIterator = eg.getAllExecutionVertices().iterator();
+		final ExecutionVertex ev11 = vertexIterator.next();
+		final ExecutionVertex ev12 = vertexIterator.next();
+		final ExecutionVertex ev21 = vertexIterator.next();
+		final ExecutionVertex ev22 = vertexIterator.next();
+
+		assertNotNull(eg.getCheckpointCoordinator());
+		assertFalse(eg.getCheckpointCoordinator().getPendingCheckpoints().isEmpty());
+
+		testMainThreadUtil.execute(() -> acknowledgeAllCheckpoints(
+			checkpointId, eg.getCheckpointCoordinator(), eg.getAllExecutionVertices().iterator()));
+
+		// verify checkpoint has been completed successfully.
+		assertEquals(1, eg.getCheckpointCoordinator().getCheckpointStore().getNumberOfRetainedCheckpoints());
+		assertEquals(checkpointId, eg.getCheckpointCoordinator().getCheckpointStore().getLatestCheckpoint(false).getCheckpointID());
+
+		testMainThreadUtil.execute(() ->ev11.getCurrentExecutionAttempt().fail(new Exception("Test Exception")));
+
+		// ensure vertex state and complete cancellation
+		assertVertexInState(ExecutionState.FAILED, ev11);
+		assertVertexInState(ExecutionState.DEPLOYING, ev12);
+		assertVertexInState(ExecutionState.CANCELING, ev21);
+		assertVertexInState(ExecutionState.DEPLOYING, ev22);
+		testMainThreadUtil.execute(() ->ev21.getCurrentExecutionAttempt().completeCancelling());
+
+		verifyCheckpointRestoredAsExpected(checkpointId, eg);
+	}
+
+	/**
+	 * Tests that the vertex partition state is reset property during failover.
+	 */
+	@Test
+	public void testStatusResettingOnRegionFailover() throws Exception {
```

Comment:
Is this test only to validate that `resetForNewExecution()` is called on the `ExecutionVertex`?

Reply:
It is to verify whether failover resetting partition state is correctly.
Will remove it as vertex resetting in adapter is covered by other cases. And resetting partition when resetting vertex is not testing changes in this PR.